spotbugs/src/main/java/edu/umd/cs/findbugs/detect/InefficientReplaceAll.java

```diff
@@ -0,0 +1,143 @@
+
+import edu.umd.cs.findbugs.BugAccumulator;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.StatelessDetector;
+import edu.umd.cs.findbugs.SystemProperties;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.classfile.MethodDescriptor;
+
+/**
+ * Find occurrences of replaceAll(String regex, String replacement) without any special regex characters. 
+ *
+ * @author Lasse Lindqvist
+ */
+public class InefficientReplaceAll extends OpcodeStackDetector implements StatelessDetector {
+    private static final boolean DEBUG = SystemProperties.getBoolean("ira.debug");
+
+    private static final List<MethodDescriptor> methods = Collections.singletonList(new MethodDescriptor("", "replaceAll",
+            "(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;"));
+
+    private final BugAccumulator bugAccumulator;
+    
+    private static List<String> regexCharList = new ArrayList<>();
+    static {
+        regexCharList.add(".");
+        regexCharList.add("\\");
+        regexCharList.add("[");
+        regexCharList.add("]");
+        regexCharList.add("{");
+        regexCharList.add("}");
+        regexCharList.add("(");
+        regexCharList.add(")");
+        regexCharList.add("<");
+        regexCharList.add(">");
+        regexCharList.add("*");
+        regexCharList.add("+");
+        regexCharList.add("-");
+        regexCharList.add("=");
+        regexCharList.add("?");
+        regexCharList.add("^");
+        regexCharList.add("$");
+        regexCharList.add("|");
+    }
+
+    public InefficientReplaceAll(BugReporter bugReporter) {
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (hasInterestingMethod(classContext.getJavaClass().getConstantPool(), methods)) {
+            classContext.getJavaClass().accept(this);
+        }
+    }
+
+    @Override
+    public void visit(Method obj) {
+        if (DEBUG) {
+            System.out.println("------------------- Analyzing " + obj.getName() + " ----------------");
+        }
+        super.visit(obj);
+    }
+
+    @Override
+    public void visit(Code obj) {
+        super.visit(obj);
+        bugAccumulator.reportAccumulatedBugs();
+
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (DEBUG) {
+            System.out.println("Opcode: " + Const.getOpcodeName(seen));
+        }
+        if (((seen == Const.INVOKEVIRTUAL) || (seen == Const.INVOKEINTERFACE)) && ("replaceAll".equals(getNameConstantOperand()))
+                && ("(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;".equals(getSigConstantOperand()))
+                && hasConstantArguments()) {
+            String firstArgument = getFirstArgument();
+            boolean found = false;
+            for (String regexChar : regexCharList) {
```

Comment:
It would likely be more efficient to implement this as a regex, instead of iterating through a list. Something like

    private static final Pattern REGEX_CHARS_PATTERN = Pattern.compile("[-.\\\\[\\]{}()<>*+=?^$|]");
    ...
    if (REGEX_CHARS_PATTERN.matcher(firstArgument).find()) {
        ...